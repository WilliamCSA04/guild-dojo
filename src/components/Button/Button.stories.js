import React from "react";
import Button from ".";

export default { title: "Button" };

export const Default = () => (
  <Button ariaLabel="Default Story">Just a button</Button>
);
