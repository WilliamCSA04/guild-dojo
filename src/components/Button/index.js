import React from "react";
import PropTypes from "prop-types";

export default function Button({
  className,
  children,
  ariaLabel,
  ariaLabelledby
}) {
  return (
    <button
      className={className}
      aria-label={ariaLabel}
      aria-labelledby={ariaLabelledby}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  ariaLabel: PropTypes.string.isRequired,
  ariaLabelledby: PropTypes.string,
  className: PropTypes.string
};

Button.defaultProps = {
  ariaLabelledby: "",
  className: ""
};
