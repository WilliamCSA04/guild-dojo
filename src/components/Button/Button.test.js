import React from "react";
import { render } from "@testing-library/react";
import Button from ".";

describe("<Button /> snapshot", () => {
  it("renders the component with default props", () => {
    const container = render(
      <Button ariaLabel="Testing" dataTestID="testing">
        Testing Button
      </Button>
    );
    expect(container.asFragment()).toMatchSnapshot();
  });

  it("renders the component with ariaLabelledby", () => {
    const container = render(
      <Button ariaLabel="Testing" ariaLabelledby="Testing" dataTestID="testing">
        Testing Button
      </Button>
    );
    expect(container.asFragment()).toMatchSnapshot();
  });
});
