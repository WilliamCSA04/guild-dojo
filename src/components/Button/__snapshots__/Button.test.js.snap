// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot renders the component with ariaLabelledby 1`] = `
<DocumentFragment>
  <button
    aria-label="Testing"
    aria-labelledby="Testing"
    class="sc-AxjAm eLfuFc"
    data-testid="testing"
    type="button"
  >
    Testing Button
  </button>
</DocumentFragment>
`;

exports[`<Button /> snapshot renders the component with default props 1`] = `
<DocumentFragment>
  <button
    aria-label="Testing"
    aria-labelledby=""
    class="sc-AxjAm eLfuFc"
    data-testid="testing"
    type="button"
  >
    Testing Button
  </button>
</DocumentFragment>
`;
